iotdb/src/test/java/org/apache/iotdb/db/engine/bufferwrite/BufferWriteProcessorNewTest.java

```diff
@@ -123,13 +126,18 @@ public void testWriteAndFlush()
       assertEquals(num, timeValuePair.getValue().getInt());
     }
     assertEquals(false, bufferwrite.isFlush());
+    long lastFlushTime = bufferwrite.getLastFlushTime();
     // flush asynchronously
     bufferwrite.flush();
-    assertEquals(true, bufferwrite.isFlush());
+    assertEquals(true, bufferwrite.getLastFlushTime() != lastFlushTime);
     assertEquals(true, bufferwrite.canBeClosed());
     // waiting for the end of flush.
-    while (bufferwrite.isFlush()) {
-      TimeUnit.SECONDS.sleep(1);
+    try {
+      bufferwrite.getFlushFuture().get(10, TimeUnit.SECONDS);
+    } catch (Exception e) {
+      //because UT uses a mock flush operation, 10 seconds should be enough.
+      Assert.fail("mock flush spends more than 10 seconds... "
+          + "Please modify the value or change a better test environment");
```

Comment:
You cannot guarantee that e is always caused by time-up, and it may also be caused by some bugs that raise exceptions during flushing, so I will recommend that you show the stacktrace as well.

Reply:
logger.error added.